src/lektorium/app.py

```diff
@@ -26,32 +35,59 @@
     )
 
 
-class RepoType(enum.Enum):
+class BaseEnum(enum.Enum):
+    @classmethod
+    def get(cls, name):
+        names = tuple(x.name for x in cls)
+        if name not in names:
+            cls_name, names = cls.__name__, ', '.join(names)
+            msg = f'Wrong {cls_name} value "{name}" should be one of {names}'
+            raise ValueError(msg)
+        return cls[name]
+
+
+class RepoType(BaseEnum):
     LIST = enum.auto()
-    LOCAL_FAKE = enum.auto()
-    LOCAL_ASYNC = enum.auto()
-    GIT = enum.auto()
+    LOCAL = enum.auto()
+
+
+class StorageType(BaseEnum):
+    FILE = FileStorage
+    GIT = GitStorage
+
 
+class ServerType(BaseEnum):
+    FAKE = FakeServer
+    ASYNC = AsyncLocalServer
+
+
+def create_app(repo_type=RepoType.LIST, auth='', repo_args=''):
+    init_logging()
 
-def create_app(repo_type=RepoType.LIST, auth=''):
-    from . import repo
     if repo_type == RepoType.LIST:
-        repo = repo.ListRepo(repo.SITES)
-    elif repo_type in (RepoType.LOCAL_FAKE, RepoType.LOCAL_ASYNC):
-        from .repo.local import LocalLektor
-        if repo_type == RepoType.LOCAL_FAKE:
-            from .repo.local import FakeServer
-            server = FakeServer()
-        else:
-            from .repo.local import AsyncLocalServer
-            server = AsyncLocalServer()
-        repo = repo.LocalRepo('gitlab', server, LocalLektor)
-    else:
-        repo = repo.GitRepo('gitlab/service')
+        if repo_args:
+            raise ValueError('LIST repo does not support arguments')
+        lektorium_repo = repo.ListRepo(repo.SITES)
+    elif repo_type == RepoType.LOCAL:
+        server_type, _, storage_config = repo_args.partition(',')
+        server = ServerType.get(server_type or 'FAKE').value()
+
+        storage_config = storage_config or 'FILE'
+        storage_type, _, storage_path = storage_config.partition('=')
+        storage_class = StorageType.get(storage_type).value
+        if not storage_path:
+            storage_path = pathlib.Path(closer(tempfile.TemporaryDirectory()))
+            storage_path = storage_class.init(storage_path)
+        storage = storage_class(storage_path)
+
+        lektorium_repo = repo.LocalRepo(storage, server, LocalLektor)
+
```

Comment:
No `else` condition. If `repo_type` of other value, `lektorium_repo` will not be initialized. Suggest adding `else` with error raising